info_extraction/src/edu/gatech/gtri/info_extraction/utils.py

```diff
@@ -0,0 +1,125 @@
+from collections import defaultdict
+from functools import reduce
+from bs4 import BeautifulSoup
+from markdown import markdown
+from pathlib import Path
+import re
+
+
+# gist:https://gist.github.com/lorey/eb15a7f3338f959a78cc3661fbc255fe
+def markdown_to_text(markdown_string):
+    """ Converts a markdown string to plaintext """
+
+    # md -> html -> text since BeautifulSoup can extract text cleanly
+    html = markdown(markdown_string)
+
+    # remove code snippets
+    html = re.sub(r'<pre>(.*?)</pre>', ' ', html)
+    html = re.sub(r'<code>(.*?)</code >', ' ', html)
+
+    # extract text
+    soup = BeautifulSoup(html, "html.parser")
+    text = ''.join(soup.findAll(text=True))
+
+    return text
+
+
+"""
+Take an array of julia code lines and extract comments
+"""
+
+
+def julia_comment_extract(julia_code_lines):
+    keeps = []
+    for line in julia_code_lines:
+        processed_line = line.lstrip(" ")
+        if processed_line.startswith("#"):
+            keeps.append(line)
```

Comment:
you could do this as a generator to save memory. Instead of building a list `keeps`, just use `yield line`

Reply:
I was writing this code under a deadline, haven't had a chance to optimize it yet but appreciate the suggestion

Reply:
That is the point of code review, to make sure the code is good.

Reply:
no, I agree, that is the point; maybe we should be using a python linter, which would make these kinds of suggestions instead of doing it manually. The linter is systematic whereas we may or may not remember to suggest the yield operator or know the internal workings of the "read" function, etc. Can we run a linter before doing code reviews in the future?